### ISSUE TITLE
Configurable optimization criterion name

### ISSUE THREAD

**Issue by petrbel** [Cognexa/cxflow-tensorflow#94]
So far, the base net specifies `loss` as the optimization criterion. It would be nice to be able to specify in in the config (and overwrite the default `loss`).

Implementing this would allow users to pretrain the net using loss `L1` and then finetune the model (in the separate training) to some other loss `L2`
